### PR TITLE
Cumulus-740 Add integration tests for reingesting a granule over API 

### PR DESCRIPTION
**Pull request by kkelly51**
Addresses [CUMULUS-740: Write an integration test demonstrating reingesting of a granule](https://bugs.earthdata.nasa.gov/browse/CUMULUS-740)

## Changes

* Moved All API integration tests to separate test location
* Added tests to reingest a granule over API
* Added tests to remove a granule from CMR over API

* Note: the reingest does work, it just takes a long time for it to show up in CMR, hence the 20 retries.

### CODE REVIEW

File: example/spec/testAPI/APISuccessSpec.js [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+  it('makes the granule available through the Cumulus API', async () => {
+    const granule = await apiTestUtils.getGranule({
+      prefix: config.stackName,
+      granuleId: inputPayload.granules[0].granuleId
+    });
+
+    expect(granule.granuleId).toEqual(inputPayload.granules[0].granuleId);
+  });
+
+  describe('reingest a granule', () => {
```
Comment (jennyhliu) [inline example/spec/testAPI/APISuccessSpec.js]: `reingest` and `applyWorkflow` are different actions, the `applyWorkflow` is for ingest-in-place which is not tested in this PR.

File: example/spec/testAPI/APISuccessSpec.js [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+      await apiTestUtils.removeFromCMR({
+        prefix: config.stackName,
+        granuleId
+      });
+      // Check that the granule was removed
+      const granuleRemoved = await waitForExist(granule.cmrLink, false, 2);
+      expect(granuleRemoved).toEqual(true);
+
+      // Reingest Granule and check that it was re-added to CMR
+      await apiTestUtils.reingestGranule({
```
Comment (jennyhliu) [inline example/spec/testAPI/APISuccessSpec.js]: Do we need to remove a granule from CMR in order to reingest or ingest-in-place a granule?

Comment (jennyhliu) [reply example/spec/testAPI/APISuccessSpec.js]: How can we tell the granule reingested/ingested-in-place is a different granule from original one?

Comment (kkelly51) [reply example/spec/testAPI/APISuccessSpec.js]: I do remove the granule from CMR (line 126) and then make sure it has been removed (line 131 and 132) before reingesting. Then I know when I find the granule in CMR again (139 adn 140) the reingest (line 135) has worked.

File: example/spec/testAPI/APISuccessSpec.js [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+
+describe('The Cumulus API', () => {
+  let workflowExecution = null;
+  const collection = { name: 'MOD09GQ', version: '006' };
+  const provider = { id: 's3_provider' };
+  const inputPayloadFilename = './spec/testAPI/testAPI.input.payload.json';
+  let inputPayload;
+  let granuleId;
+
+  beforeAll(async () => {
```
Comment (jennyhliu) [inline example/spec/testAPI/APISuccessSpec.js]: I'm wondering if we should put the api tests under spec/ingestGranule/IngestGranuleSuccessSpec.js.  The setup and workflow are the same.

**Pull request merged**